File: app/types/cron.ts
import { Category } from "@/types/category-model"
import { User } from "@/types/index"

export type CronProduct = {
    product: string
    user: string
    category: Category
    id?: string | null
}

export type CreateCronProductRequest = {
    category: string
    user: User
    product: string
}

export type CreateCronProductResponse = CronProduct

export type UpdateCronProductCategoryRequest = {
    category: string
}

export type UpdateCronProductCategoryResponse = CronProduct

export type GetAllCronProductsResponse = CronProduct[]

export type GetCronProductsByUserResponse = CronProduct[]

export type DeleteCronProductResponse = {
    id: string
    message: string
}

File: services/shared/contracts/cron.go
package contracts

import "shopping-list/shared/models"

type CreateCronProductRequest struct {
	Category string `json:"category" validate:"required"`
	User     string `json:"user" validate:"required"`
	Product  string `json:"product" validate:"required"`
}
type CreateCronProductResponse models.CronProduct

type UpdateCronProductCategoryRequest struct {
	Category string `json:"category" validate:"required"`
}

type UpdateCronProductCategoryResponse models.CronProduct

type GetAllCronProductsResponse []models.CronProduct

type GetCronProductsByUserResponse []models.CronProduct

type DeleteCronProductResponse struct {
	Id      string `json:"id"`
	Message string `json:"message,omitempty"`
}
